<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<meta name="date" content="2019-06-25" />

<title>Clustering</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Oz Single Cells 2019</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lazappi/OzSingleCells2019">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Clustering</h1>
<h4 class="date">2019-06-25</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-06-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>OzSingleCells2019/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190619code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190619)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190619code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190619)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpgithubcomlazappiOzSingleCells2019tree25950571da44bb4ee287248061daa78c701cbf09targetblank2595057a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="http://github.com/lazappi/OzSingleCells2019/tree/25950571da44bb4ee287248061daa78c701cbf09" target="_blank">2595057</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpgithubcomlazappiOzSingleCells2019tree25950571da44bb4ee287248061daa78c701cbf09targetblank2595057a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    analysis/cache/
    Ignored:    data/._antibody_genes.tsv
    Ignored:    data/._antibody_genes.txt
    Ignored:    docs/.DS_Store
    Ignored:    docs/._.DS_Store
    Ignored:    output/03-comparison.Rmd/
    Ignored:    packrat/lib-R/
    Ignored:    packrat/lib-ext/
    Ignored:    packrat/lib/
    Ignored:    packrat/src/

Untracked files:
    Untracked:  data/04-clustered.Rds
    Untracked:  docs/figure/04-clustering.Rmd/
    Untracked:  output/04-clustering.Rmd/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="http://github.com/lazappi/OzSingleCells2019/blob/25950571da44bb4ee287248061daa78c701cbf09/analysis/04-clustering.Rmd" target="_blank">2595057</a>
</td>
<td>
Luke Zappia
</td>
<td>
2019-06-25
</td>
<td>
Add clustering of RNA data
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>#### LIBRARIES ####
# Package conflicts
library(&quot;conflicted&quot;)

# Single-cell
library(&quot;SingleCellExperiment&quot;)
library(&quot;scran&quot;)
library(&quot;scater&quot;)

# Plotting
library(&quot;clustree&quot;)
library(&quot;ggforce&quot;)

# Bioconductor
library(&quot;BiocSingular&quot;)

# File paths
library(&quot;fs&quot;)
library(&quot;here&quot;)

# Presentation
library(&quot;knitr&quot;)
library(&quot;jsonlite&quot;)

# Tidyverse
library(&quot;tidyverse&quot;)

### CONFLICT PREFERENCES ####
conflict_prefer(&quot;path&quot;, &quot;fs&quot;)
conflict_prefer(&quot;mutate&quot;, &quot;dplyr&quot;)

### SOURCE FUNCTIONS ####
source(here(&quot;R/output.R&quot;))

### OUTPUT DIRECTORY ####
OUT_DIR &lt;- here(&quot;output&quot;, DOCNAME)
dir_create(OUT_DIR)

#### SET GGPLOT THEME ####
theme_set(theme_minimal())

#### SET PATHS ####
source(here(&quot;R/set_paths.R&quot;))</code></pre>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>In document we are going to perform a standard clustering analysis using just the RNA-seq data. This analysis is based on the <a href="https://bioconductor.org/packages/release/workflows/html/simpleSingleCell.html">“simpleSingleCell”</a> workflow.</p>
<pre class="r"><code>if (all(file_exists(c(PATHS$sce_qc)))) {
    sce &lt;- read_rds(PATHS$sce_qc)
} else {
    stop(&quot;Filtered dataset is missing. &quot;,
         &quot;Please run &#39;02-quality-control.Rmd&#39; first.&quot;,
         call. = FALSE)
}

col_data &lt;- as.data.frame(colData(sce))</code></pre>
</div>
<div id="normalisation" class="section level1">
<h1>Normalisation</h1>
<div id="clusters" class="section level2">
<h2>Clusters</h2>
<p>Pre-cluster cells to avoid the assumption that most genes are non-DE.</p>
<pre class="r"><code>set.seed(1)
clusters &lt;- quickCluster(sce, use.ranks = FALSE, BSPARAM = IrlbaParam())
col_data$NormCluster &lt;- clusters

kable(table(clusters), col.names = c(&quot;Cluster&quot;, &quot;Size&quot;))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Cluster</th>
<th align="right">Size</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="right">322</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="right">224</td>
</tr>
<tr class="odd">
<td align="left">3</td>
<td align="right">115</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="right">202</td>
</tr>
<tr class="odd">
<td align="left">5</td>
<td align="right">225</td>
</tr>
<tr class="even">
<td align="left">6</td>
<td align="right">200</td>
</tr>
<tr class="odd">
<td align="left">7</td>
<td align="right">284</td>
</tr>
<tr class="even">
<td align="left">8</td>
<td align="right">180</td>
</tr>
<tr class="odd">
<td align="left">9</td>
<td align="right">169</td>
</tr>
<tr class="even">
<td align="left">10</td>
<td align="right">150</td>
</tr>
<tr class="odd">
<td align="left">11</td>
<td align="right">107</td>
</tr>
</tbody>
</table>
</div>
<div id="size-factors" class="section level2">
<h2>Size factors</h2>
<p>Calculate <code>scran</code> doconvolution size factors.</p>
<pre class="r"><code>sce &lt;- computeSumFactors(sce, min.mean = 0.1, cluster = clusters)
col_data$SizeFactor &lt;- sizeFactors(sce)

ggplot(col_data,
       aes(x = total_counts, y = SizeFactor, colour = NormCluster, group = 1)) +
    geom_point() +
    geom_smooth(method = &quot;lm&quot;) +
    scale_x_log10() +
    scale_y_log10() +
    labs(
        title = &quot;Correlation of size factors&quot;,
        x = &quot;Total counts&quot;,
        y = &quot;Size factor&quot;,
        colour = &quot;Normalisation\ncluster&quot;
    )</code></pre>
<p><img src="figure/04-clustering.Rmd/norm-size-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sce &lt;- normalize(sce)</code></pre>
</div>
</div>
<div id="mean-variance" class="section level1">
<h1>Mean-variance</h1>
<div id="fit" class="section level2">
<h2>Fit</h2>
<p>Fit the mean-variance relationship</p>
<pre class="r"><code>tech_trend &lt;- makeTechTrend(x = sce)
fit &lt;- trendVar(sce, use.spikes = FALSE, loess.args = list(span = 0.05))

plot_data &lt;- tibble(
    Mean = fit$means,
    Var = fit$vars,
    Trend = fit$trend(fit$means),
    TechTrend = tech_trend(fit$means)
)

ggplot(plot_data, aes(x = Mean)) +
    geom_point(aes(y = Var)) +
    geom_line(aes(y = Trend), colour = &quot;blue&quot;) +
    geom_line(aes(y = TechTrend), colour = &quot;red&quot;) +
    annotate(&quot;text&quot;, x = 7, y = 2, label = &quot;Fitted trend&quot;, colour = &quot;blue&quot;) +
    annotate(&quot;text&quot;, x = 7, y = 1.8, label = &quot;Poisson noise&quot;, colour = &quot;red&quot;) +
    labs(
        title = &quot;Mean-variance relationship&quot;,
        x = &quot;Mean log-expression&quot;,
        y = &quot;Variance of log-expression&quot;
    )</code></pre>
<p><img src="figure/04-clustering.Rmd/fit-mean-var-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="decompose" class="section level2">
<h2>Decompose</h2>
<p>Decompose variance into technical and biological factors based on the technical Poisson trend.</p>
<pre class="r"><code>fit$trend &lt;- tech_trend
decomposed &lt;- decomposeVar(fit = fit)
top_dec &lt;- decomposed[order(decomposed$bio, decreasing = TRUE), ] 

kable(head(rownames_to_column(as.data.frame(top_dec), &quot;Gene&quot;), n = 10))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Gene</th>
<th align="right">mean</th>
<th align="right">total</th>
<th align="right">bio</th>
<th align="right">tech</th>
<th align="right">p.value</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CD74</td>
<td align="right">2.641586</td>
<td align="right">5.256015</td>
<td align="right">4.685522</td>
<td align="right">0.5704924</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">HLA-DRA</td>
<td align="right">1.611305</td>
<td align="right">4.648042</td>
<td align="right">3.829236</td>
<td align="right">0.8188056</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">JUNB</td>
<td align="right">3.245805</td>
<td align="right">2.820314</td>
<td align="right">2.435900</td>
<td align="right">0.3844141</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">FOS</td>
<td align="right">2.111452</td>
<td align="right">2.934361</td>
<td align="right">2.203540</td>
<td align="right">0.7308210</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">HLA-DPA1</td>
<td align="right">1.405453</td>
<td align="right">2.833084</td>
<td align="right">2.010756</td>
<td align="right">0.8223286</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">HLA-DPB1</td>
<td align="right">1.379264</td>
<td align="right">2.563782</td>
<td align="right">1.742731</td>
<td align="right">0.8210517</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">IL7R</td>
<td align="right">1.721153</td>
<td align="right">2.496847</td>
<td align="right">1.688833</td>
<td align="right">0.8080145</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">RGS1</td>
<td align="right">1.590717</td>
<td align="right">2.352885</td>
<td align="right">1.532707</td>
<td align="right">0.8201778</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">ZFP36L2</td>
<td align="right">3.363269</td>
<td align="right">1.857358</td>
<td align="right">1.504431</td>
<td align="right">0.3529270</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">HLA-DQB1</td>
<td align="right">1.171683</td>
<td align="right">2.285178</td>
<td align="right">1.489623</td>
<td align="right">0.7955555</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>plotExpression(sce, features = rownames(top_dec)[1:10]) +
    ggtitle(&quot;Expression of biologically variable genes&quot;) +
    theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/decompose-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="dimensionality-reduction" class="section level1">
<h1>Dimensionality reduction</h1>
<div id="pca" class="section level2">
<h2>PCA</h2>
<p>PCA is performed on the dataset and the Poisson technical variance trend is used to select the top components to use.</p>
<pre class="r"><code>set.seed(1)
sce &lt;- denoisePCA(sce, technical = tech_trend, BSPARAM = IrlbaParam())
n_pcs &lt;- ncol(reducedDim(sce, &quot;PCA&quot;))

plot_data &lt;- tibble(
    PC = seq_along(attr(reducedDim(sce), &quot;percentVar&quot;)),
    PercentVar = attr(reducedDim(sce), &quot;percentVar&quot;)
) %&gt;%
    mutate(Selected = PC &lt;= n_pcs)

ggplot(plot_data, aes(x = PC, y = PercentVar, colour = Selected)) +
    geom_point() +
    scale_colour_manual(values = c(&quot;grey40&quot;, &quot;red&quot;)) +
    labs(
        title = &quot;PC variance&quot;,
        x = &quot;Principal component&quot;,
        y = &quot;Proportion of variance explained&quot;
    )</code></pre>
<p><img src="figure/04-clustering.Rmd/denoise-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotPCA(sce, ncomponents = 3, colour_by = &quot;log10_total_features_by_counts&quot;)</code></pre>
<p><img src="figure/04-clustering.Rmd/denoise-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>Here we select the first 24 components.</p>
</div>
<div id="t-sne" class="section level2">
<h2>t-SNE</h2>
<pre class="r"><code>set.seed(1)
sce &lt;- runTSNE(sce, use_dimred = &quot;PCA&quot;, perplexity = 40)
plotTSNE(sce, colour_by = &quot;log10_total_features_by_counts&quot;)</code></pre>
<p><img src="figure/04-clustering.Rmd/t-SNE-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="clustering" class="section level1">
<h1>Clustering</h1>
<p>Cluster cells using the implementation in <code>Seurat</code> because it has a resolution parameter.</p>
<pre class="r"><code>snn_mat &lt;- Seurat::FindNeighbors(reducedDim(sce, &quot;PCA&quot;))$snn

resolutions &lt;- seq(0, 1, 0.1)
for (res in resolutions) {
    clusters &lt;- Seurat:::RunModularityClustering(snn_mat, resolution = res)
    col_data[[paste0(&quot;ClusterRes&quot;, res)]] &lt;- factor(clusters)
}

colData(sce) &lt;- DataFrame(col_data)</code></pre>
<div id="dimensionality-reduction-1" class="section level2 tabset">
<h2>Dimensionality reduction</h2>
<p>Dimensionality reduction plots showing clusters at different resolutions.</p>
<div id="pca-1" class="section level3 tabset">
<h3>PCA</h3>
<pre class="r"><code>src_list &lt;- lapply(resolutions, function(res) {
    src &lt;- c(
        &quot;#### Res {{res}} {.unnumbered}&quot;,
        &quot;```{r res-pca-{{res}}}&quot;,
        &quot;plotPCA(sce, colour_by = &#39;ClusterRes{{res}}&#39;) + theme_minimal()&quot;,
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="res-0" class="section level4 unnumbered">
<h4>Res 0</h4>
<pre class="r"><code>plotPCA(sce, colour_by = &#39;ClusterRes0&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-pca-0-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-0.1" class="section level4 unnumbered">
<h4>Res 0.1</h4>
<pre class="r"><code>plotPCA(sce, colour_by = &#39;ClusterRes0.1&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-pca-0.1-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-0.2" class="section level4 unnumbered">
<h4>Res 0.2</h4>
<pre class="r"><code>plotPCA(sce, colour_by = &#39;ClusterRes0.2&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-pca-0.2-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-0.3" class="section level4 unnumbered">
<h4>Res 0.3</h4>
<pre class="r"><code>plotPCA(sce, colour_by = &#39;ClusterRes0.3&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-pca-0.3-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-0.4" class="section level4 unnumbered">
<h4>Res 0.4</h4>
<pre class="r"><code>plotPCA(sce, colour_by = &#39;ClusterRes0.4&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-pca-0.4-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-0.5" class="section level4 unnumbered">
<h4>Res 0.5</h4>
<pre class="r"><code>plotPCA(sce, colour_by = &#39;ClusterRes0.5&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-pca-0.5-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-0.6" class="section level4 unnumbered">
<h4>Res 0.6</h4>
<pre class="r"><code>plotPCA(sce, colour_by = &#39;ClusterRes0.6&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-pca-0.6-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-0.7" class="section level4 unnumbered">
<h4>Res 0.7</h4>
<pre class="r"><code>plotPCA(sce, colour_by = &#39;ClusterRes0.7&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-pca-0.7-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-0.8" class="section level4 unnumbered">
<h4>Res 0.8</h4>
<pre class="r"><code>plotPCA(sce, colour_by = &#39;ClusterRes0.8&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-pca-0.8-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-0.9" class="section level4 unnumbered">
<h4>Res 0.9</h4>
<pre class="r"><code>plotPCA(sce, colour_by = &#39;ClusterRes0.9&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-pca-0.9-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-1" class="section level4 unnumbered">
<h4>Res 1</h4>
<pre class="r"><code>plotPCA(sce, colour_by = &#39;ClusterRes1&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-pca-1-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="t-sne-1" class="section level3 tabset">
<h3>t-SNE</h3>
<pre class="r"><code>src_list &lt;- lapply(resolutions, function(res) {
    src &lt;- c(
        &quot;#### Res {{res}} {.unnumbered}&quot;,
        &quot;```{r res-tSNE-{{res}}}&quot;,
        &quot;plotTSNE(sce, colour_by = &#39;ClusterRes{{res}}&#39;) + theme_minimal()&quot;,  
        &quot;```&quot;,
        &quot;&quot;
    )
    knit_expand(text = src)
})
out &lt;- knit_child(text = unlist(src_list), options = list(cache = FALSE))</code></pre>
<div id="res-0-1" class="section level4 unnumbered">
<h4>Res 0</h4>
<pre class="r"><code>plotTSNE(sce, colour_by = &#39;ClusterRes0&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-tSNE-0-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-0.1-1" class="section level4 unnumbered">
<h4>Res 0.1</h4>
<pre class="r"><code>plotTSNE(sce, colour_by = &#39;ClusterRes0.1&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-tSNE-0.1-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-0.2-1" class="section level4 unnumbered">
<h4>Res 0.2</h4>
<pre class="r"><code>plotTSNE(sce, colour_by = &#39;ClusterRes0.2&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-tSNE-0.2-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-0.3-1" class="section level4 unnumbered">
<h4>Res 0.3</h4>
<pre class="r"><code>plotTSNE(sce, colour_by = &#39;ClusterRes0.3&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-tSNE-0.3-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-0.4-1" class="section level4 unnumbered">
<h4>Res 0.4</h4>
<pre class="r"><code>plotTSNE(sce, colour_by = &#39;ClusterRes0.4&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-tSNE-0.4-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-0.5-1" class="section level4 unnumbered">
<h4>Res 0.5</h4>
<pre class="r"><code>plotTSNE(sce, colour_by = &#39;ClusterRes0.5&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-tSNE-0.5-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-0.6-1" class="section level4 unnumbered">
<h4>Res 0.6</h4>
<pre class="r"><code>plotTSNE(sce, colour_by = &#39;ClusterRes0.6&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-tSNE-0.6-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-0.7-1" class="section level4 unnumbered">
<h4>Res 0.7</h4>
<pre class="r"><code>plotTSNE(sce, colour_by = &#39;ClusterRes0.7&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-tSNE-0.7-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-0.8-1" class="section level4 unnumbered">
<h4>Res 0.8</h4>
<pre class="r"><code>plotTSNE(sce, colour_by = &#39;ClusterRes0.8&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-tSNE-0.8-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-0.9-1" class="section level4 unnumbered">
<h4>Res 0.9</h4>
<pre class="r"><code>plotTSNE(sce, colour_by = &#39;ClusterRes0.9&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-tSNE-0.9-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="res-1-1" class="section level4 unnumbered">
<h4>Res 1</h4>
<pre class="r"><code>plotTSNE(sce, colour_by = &#39;ClusterRes1&#39;) + theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/res-tSNE-1-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="clustering-trees" class="section level2 tabset">
<h2>Clustering trees</h2>
<p>Clustering trees show the relationship between clusterings at adjacent resolutions. Each cluster is represented as a node in a graph and the edges show the overlap between clusters.</p>
<div id="standard" class="section level3">
<h3>Standard</h3>
<p>Coloured by clustering resolution.</p>
<pre class="r"><code>clustree(sce, prefix = &quot;ClusterRes&quot;)</code></pre>
<p><img src="figure/04-clustering.Rmd/clustree-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="stability" class="section level3">
<h3>Stability</h3>
<p>Coloured by the SC3 stability metric.</p>
<pre class="r"><code>clustree(sce, prefix = &quot;ClusterRes&quot;, node_colour = &quot;sc3_stability&quot;)</code></pre>
<p><img src="figure/04-clustering.Rmd/clustree-stability-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="selection" class="section level2">
<h2>Selection</h2>
<pre class="r"><code>res &lt;- 0.8
col_data$Cluster &lt;- col_data[[paste0(&quot;ClusterRes&quot;, res)]]
colData(sce) &lt;- DataFrame(col_data)
n_clusts &lt;- length(unique(col_data$Cluster))</code></pre>
<p>Based on these plots we will use a resolution of 0.8 which gives us 13 clusters.</p>
</div>
</div>
<div id="validation" class="section level1 tabset">
<h1>Validation</h1>
<p>To validate the clusters we will repeat some of our quality control plots separated by cluster. At this stage we just want to check that none of the clusters are obviously the result of technical factors.</p>
<div id="cluster" class="section level2 tabset">
<h2>Cluster</h2>
<p>Clusters assigned by <code>Seurat</code>.</p>
<div id="count" class="section level3">
<h3>Count</h3>
<pre class="r"><code>ggplot(col_data, aes(x = Cluster, fill = Cluster)) +
    geom_bar()</code></pre>
<p><img src="figure/04-clustering.Rmd/val-cluster-count-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="pca-2" class="section level3">
<h3>PCA</h3>
<pre class="r"><code>plotPCA(sce, colour_by = &quot;Cluster&quot;, point_alpha = 1) +
    scale_fill_discrete() +
    theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/val-cluster-pca-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="t-sne-2" class="section level3">
<h3>t-SNE</h3>
<pre class="r"><code>plotTSNE(sce, colour_by = &quot;Cluster&quot;, point_alpha = 1) +
    scale_fill_discrete() +
    theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/val-cluster-tSNE-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="cell-cycle" class="section level2 tabset">
<h2>Cell cycle</h2>
<p>Cell cycle phases assigned by <code>scran</code>.</p>
<div id="count-1" class="section level3">
<h3>Count</h3>
<pre class="r"><code>ggplot(col_data, aes(x = Cluster, fill = CellCycle)) +
    geom_bar()</code></pre>
<p><img src="figure/04-clustering.Rmd/val-cycle-count-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="proportion" class="section level3">
<h3>Proportion</h3>
<pre class="r"><code>plot_data &lt;- col_data %&gt;%
    group_by(Cluster, CellCycle) %&gt;%
    summarise(Count = n()) %&gt;%
    mutate(Prop = Count / sum(Count))

ggplot(plot_data, aes(x = Cluster, y = Prop, fill = CellCycle)) +
    geom_col() +
    ylab(&quot;Proportion of cluster&quot;)</code></pre>
<p><img src="figure/04-clustering.Rmd/val-cycle-prop-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="pca-3" class="section level3">
<h3>PCA</h3>
<pre class="r"><code>plotPCA(sce, colour_by = &quot;CellCycle&quot;, point_alpha = 1) +
    scale_fill_discrete() +
    theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/val-cycle-pca-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="t-sne-3" class="section level3">
<h3>t-SNE</h3>
<pre class="r"><code>plotTSNE(sce, colour_by = &quot;CellCycle&quot;, point_alpha = 1) +
    scale_fill_discrete() +
    theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/val-cycle-tSNE-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="total-counts" class="section level2 tabset">
<h2>Total counts</h2>
<p>Total counts per cell.</p>
<div id="distribution" class="section level3">
<h3>Distribution</h3>
<pre class="r"><code>ggplot(col_data, aes(x = Cluster, y = log10_total_counts, colour = Cluster)) +
    geom_violin() +
    geom_sina(size = 0.5) +
    theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/04-clustering.Rmd/val-counts-dist-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="pca-4" class="section level3">
<h3>PCA</h3>
<pre class="r"><code>plotPCA(sce, colour_by = &quot;log10_total_counts&quot;, point_alpha = 1) +
    scale_fill_viridis_c() +
    theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/val-counts-pca-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="t-sne-4" class="section level3">
<h3>t-SNE</h3>
<pre class="r"><code>plotTSNE(sce, colour_by = &quot;log10_total_counts&quot;, point_alpha = 1) +
    scale_fill_viridis_c()+
    theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/val-counts-tSNE-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="total-features" class="section level2 tabset">
<h2>Total features</h2>
<p>Total number of expressed features per cell.</p>
<div id="distribution-1" class="section level3">
<h3>Distribution</h3>
<pre class="r"><code>ggplot(col_data,
       aes(x = Cluster, y = log10_total_features_by_counts, colour = Cluster)) +
    geom_violin() +
    geom_sina(size = 0.5) +
    theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/04-clustering.Rmd/val-features-dist-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="pca-5" class="section level3">
<h3>PCA</h3>
<pre class="r"><code>plotPCA(sce, colour_by = &quot;log10_total_features_by_counts&quot;, point_alpha = 1) +
    scale_fill_viridis_c() +
    theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/val-features-pca-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="t-sne-5" class="section level3">
<h3>t-SNE</h3>
<pre class="r"><code>plotTSNE(sce, colour_by = &quot;log10_total_features_by_counts&quot;, point_alpha = 1) +
    scale_fill_viridis_c()+
    theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/val-features-tSNE-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="mitochondrial-genes" class="section level2 tabset">
<h2>Mitochondrial genes</h2>
<p>Percentage of counts assigned to mitochondrial genes per cell.</p>
<div id="distribution-2" class="section level3">
<h3>Distribution</h3>
<pre class="r"><code>ggplot(col_data, aes(x = Cluster, y = pct_counts_MT, colour = Cluster)) +
    geom_violin() +
    geom_sina(size = 0.5) +
    theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/04-clustering.Rmd/val-mt-dist-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="pca-6" class="section level3">
<h3>PCA</h3>
<pre class="r"><code>plotPCA(sce, colour_by = &quot;pct_counts_MT&quot;, point_alpha = 1) +
    scale_fill_viridis_c() +
    theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/val-mt-pca-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="t-sne-6" class="section level3">
<h3>t-SNE</h3>
<pre class="r"><code>plotTSNE(sce, colour_by = &quot;pct_counts_MT&quot;, point_alpha = 1) +
    scale_fill_viridis_c()+
    theme_minimal()</code></pre>
<p><img src="figure/04-clustering.Rmd/val-mt-tSNE-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<div id="parameters" class="section level2">
<h2>Parameters</h2>
<p>This table describes parameters used and set in this document.</p>
<pre class="r"><code>params &lt;- list(
    list(
        Parameter = &quot;n_pcs&quot;,
        Value = n_pcs,
        Description = &quot;Selected number of principal components for clustering&quot;
    ),
    list(
        Parameter = &quot;resolutions&quot;,
        Value = resolutions,
        Description = &quot;Range of possible clustering resolutions&quot;
    ),
    list(
        Parameter = &quot;res&quot;,
        Value = res,
        Description = &quot;Selected resolution parameter for clustering&quot;
    ),
    list(
        Parameter = &quot;n_clusts&quot;,
        Value = n_clusts,
        Description = &quot;Number of clusters produced by selected resolution&quot;
    )
)

params &lt;- toJSON(params, pretty = TRUE)
kable(fromJSON(params))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Parameter</th>
<th align="left">Value</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">n_pcs</td>
<td align="left">24</td>
<td align="left">Selected number of principal components for clustering</td>
</tr>
<tr class="even">
<td align="left">resolutions</td>
<td align="left">c(0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1)</td>
<td align="left">Range of possible clustering resolutions</td>
</tr>
<tr class="odd">
<td align="left">res</td>
<td align="left">0.8</td>
<td align="left">Selected resolution parameter for clustering</td>
</tr>
<tr class="even">
<td align="left">n_clusts</td>
<td align="left">13</td>
<td align="left">Number of clusters produced by selected resolution</td>
</tr>
</tbody>
</table>
</div>
<div id="output-files" class="section level2">
<h2>Output files</h2>
<p>This table describes the output files produced by this document. Right click and <em>Save Link As…</em> to download the results.</p>
<pre class="r"><code>write_rds(sce, PATHS$sce_clust, compress = &quot;bz&quot;, compression = 9)
write_lines(params, path(OUT_DIR, &quot;parameters.json&quot;))

kable(data.frame(
    File = c(
        download_link(&quot;parameters.json&quot;, OUT_DIR)
    ),
    Description = c(
        &quot;Parameters set and used in this analysis&quot;
    )
))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">File</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="http://github.com/lazappi/OzSingleCells2019/raw/master/output/04-clustering.Rmd/parameters.json">parameters.json</a></td>
<td align="left">Parameters set and used in this analysis</td>
</tr>
</tbody>
</table>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessioninfo::session_info()</code></pre>
<pre><code>─ Session info ──────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.0 (2019-04-26)
 os       CentOS release 6.7 (Final)  
 system   x86_64, linux-gnu           
 ui       X11                         
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       Australia/Melbourne         
 date     2019-06-25                  

─ Packages ──────────────────────────────────────────────────────────────
 ! package              * version   date       lib source        
   ape                    5.3       2019-03-17 [1] CRAN (R 3.6.0)
   assertthat             0.2.1     2019-03-21 [1] CRAN (R 3.6.0)
   backports              1.1.4     2019-04-10 [1] CRAN (R 3.6.0)
   beeswarm               0.2.3     2016-04-25 [1] CRAN (R 3.6.0)
   bibtex                 0.4.2     2017-06-30 [1] CRAN (R 3.6.0)
   Biobase              * 2.44.0    2019-05-02 [1] Bioconductor  
   BiocGenerics         * 0.30.0    2019-05-02 [1] Bioconductor  
   BiocNeighbors          1.2.0     2019-05-02 [1] Bioconductor  
   BiocParallel         * 1.18.0    2019-05-03 [1] Bioconductor  
   BiocSingular         * 1.0.0     2019-05-02 [1] Bioconductor  
   bitops                 1.0-6     2013-08-17 [1] CRAN (R 3.6.0)
   broom                  0.5.2     2019-04-07 [1] CRAN (R 3.6.0)
   caTools                1.17.1.2  2019-03-06 [1] CRAN (R 3.6.0)
   cellranger             1.1.0     2016-07-27 [1] CRAN (R 3.6.0)
   checkmate              1.9.3     2019-05-03 [1] CRAN (R 3.6.0)
   cli                    1.1.0     2019-03-19 [1] CRAN (R 3.6.0)
 P cluster                2.0.9     2019-05-01 [5] CRAN (R 3.6.0)
   clustree             * 0.4.0     2019-04-18 [1] CRAN (R 3.6.0)
 P codetools              0.2-16    2018-12-24 [5] CRAN (R 3.6.0)
   colorspace             1.4-1     2019-03-18 [1] CRAN (R 3.6.0)
   conflicted           * 1.0.3     2019-05-01 [1] CRAN (R 3.6.0)
   cowplot                0.9.4     2019-01-08 [1] CRAN (R 3.6.0)
   crayon                 1.3.4     2017-09-16 [1] CRAN (R 3.6.0)
   data.table             1.12.2    2019-04-07 [1] CRAN (R 3.6.0)
   DelayedArray         * 0.10.0    2019-05-02 [1] Bioconductor  
   DelayedMatrixStats     1.6.0     2019-05-02 [1] Bioconductor  
   digest                 0.6.19    2019-05-20 [1] CRAN (R 3.6.0)
   dplyr                * 0.8.1     2019-05-14 [1] CRAN (R 3.6.0)
   dqrng                  0.2.1     2019-05-17 [1] CRAN (R 3.6.0)
   dynamicTreeCut         1.63-1    2016-03-11 [1] CRAN (R 3.6.0)
   edgeR                  3.26.4    2019-05-27 [1] Bioconductor  
   evaluate               0.14      2019-05-28 [1] CRAN (R 3.6.0)
   farver                 1.1.0     2018-11-20 [1] CRAN (R 3.6.0)
   fitdistrplus           1.0-14    2019-01-23 [1] CRAN (R 3.6.0)
   forcats              * 0.4.0     2019-02-17 [1] CRAN (R 3.6.0)
   fs                   * 1.3.1     2019-05-06 [1] CRAN (R 3.6.0)
   future                 1.13.0    2019-05-08 [1] CRAN (R 3.6.0)
   future.apply           1.3.0     2019-06-18 [1] CRAN (R 3.6.0)
   gbRd                   0.4-11    2012-10-01 [1] CRAN (R 3.6.0)
   gdata                  2.18.0    2017-06-06 [1] CRAN (R 3.6.0)
   generics               0.0.2     2018-11-29 [1] CRAN (R 3.6.0)
   GenomeInfoDb         * 1.20.0    2019-05-02 [1] Bioconductor  
   GenomeInfoDbData       1.2.1     2019-06-19 [1] Bioconductor  
   GenomicRanges        * 1.36.0    2019-05-02 [1] Bioconductor  
   ggbeeswarm             0.6.0     2017-08-07 [1] CRAN (R 3.6.0)
   ggforce              * 0.2.2     2019-04-23 [1] CRAN (R 3.6.0)
   ggplot2              * 3.2.0     2019-06-16 [1] CRAN (R 3.6.0)
   ggraph               * 1.0.2     2018-07-07 [1] CRAN (R 3.6.0)
   ggrepel                0.8.1     2019-05-07 [1] CRAN (R 3.6.0)
   ggridges               0.5.1     2018-09-27 [1] CRAN (R 3.6.0)
   git2r                  0.25.2    2019-03-19 [1] CRAN (R 3.6.0)
   globals                0.12.4    2018-10-11 [1] CRAN (R 3.6.0)
   glue                   1.3.1     2019-03-12 [1] CRAN (R 3.6.0)
   gplots                 3.0.1.1   2019-01-27 [1] CRAN (R 3.6.0)
   gridExtra              2.3       2017-09-09 [1] CRAN (R 3.6.0)
   gtable                 0.3.0     2019-03-25 [1] CRAN (R 3.6.0)
   gtools                 3.8.1     2018-06-26 [1] CRAN (R 3.6.0)
   haven                  2.1.0     2019-02-19 [1] CRAN (R 3.6.0)
   here                 * 0.1       2017-05-28 [1] CRAN (R 3.6.0)
   highr                  0.8       2019-03-20 [1] CRAN (R 3.6.0)
   hms                    0.4.2     2018-03-10 [1] CRAN (R 3.6.0)
   htmltools              0.3.6     2017-04-28 [1] CRAN (R 3.6.0)
   htmlwidgets            1.3       2018-09-30 [1] CRAN (R 3.6.0)
   httr                   1.4.0     2018-12-11 [1] CRAN (R 3.6.0)
   ica                    1.0-2     2018-05-24 [1] CRAN (R 3.6.0)
   igraph                 1.2.4.1   2019-04-22 [1] CRAN (R 3.6.0)
   IRanges              * 2.18.1    2019-05-31 [1] Bioconductor  
   irlba                  2.3.3     2019-02-05 [1] CRAN (R 3.6.0)
   jsonlite             * 1.6       2018-12-07 [1] CRAN (R 3.6.0)
 P KernSmooth             2.23-15   2015-06-29 [5] CRAN (R 3.6.0)
   knitr                * 1.23      2019-05-18 [1] CRAN (R 3.6.0)
   labeling               0.3       2014-08-23 [1] CRAN (R 3.6.0)
 P lattice                0.20-38   2018-11-04 [5] CRAN (R 3.6.0)
   lazyeval               0.2.2     2019-03-15 [1] CRAN (R 3.6.0)
   limma                  3.40.2    2019-05-17 [1] Bioconductor  
   listenv                0.7.0     2018-01-21 [1] CRAN (R 3.6.0)
   lmtest                 0.9-37    2019-04-30 [1] CRAN (R 3.6.0)
   locfit                 1.5-9.1   2013-04-20 [1] CRAN (R 3.6.0)
   lsei                   1.2-0     2017-10-23 [1] CRAN (R 3.6.0)
   lubridate              1.7.4     2018-04-11 [1] CRAN (R 3.6.0)
   magrittr               1.5       2014-11-22 [1] CRAN (R 3.6.0)
 P MASS                   7.3-51.4  2019-04-26 [5] CRAN (R 3.6.0)
 P Matrix                 1.2-17    2019-03-22 [5] CRAN (R 3.6.0)
   matrixStats          * 0.54.0    2018-07-23 [1] CRAN (R 3.6.0)
   memoise                1.1.0     2017-04-21 [1] CRAN (R 3.6.0)
   metap                  1.1       2019-02-06 [1] CRAN (R 3.6.0)
   modelr                 0.1.4     2019-02-18 [1] CRAN (R 3.6.0)
   munsell                0.5.0     2018-06-12 [1] CRAN (R 3.6.0)
 P nlme                   3.1-139   2019-04-09 [5] CRAN (R 3.6.0)
   npsurv                 0.4-0     2017-10-14 [1] CRAN (R 3.6.0)
   pbapply                1.4-0     2019-02-05 [1] CRAN (R 3.6.0)
   pillar                 1.4.1     2019-05-28 [1] CRAN (R 3.6.0)
   pkgconfig              2.0.2     2018-08-16 [1] CRAN (R 3.6.0)
   plotly                 4.9.0     2019-04-10 [1] CRAN (R 3.6.0)
   plyr                   1.8.4     2016-06-08 [1] CRAN (R 3.6.0)
   png                    0.1-7     2013-12-03 [1] CRAN (R 3.6.0)
   polyclip               1.10-0    2019-03-14 [1] CRAN (R 3.6.0)
   purrr                * 0.3.2     2019-03-15 [1] CRAN (R 3.6.0)
   R.methodsS3            1.7.1     2016-02-16 [1] CRAN (R 3.6.0)
   R.oo                   1.22.0    2018-04-22 [1] CRAN (R 3.6.0)
   R.utils                2.9.0     2019-06-13 [1] CRAN (R 3.6.0)
   R6                     2.4.0     2019-02-14 [1] CRAN (R 3.6.0)
   RANN                   2.6.1     2019-01-08 [1] CRAN (R 3.6.0)
   RColorBrewer           1.1-2     2014-12-07 [1] CRAN (R 3.6.0)
   Rcpp                   1.0.1     2019-03-17 [1] CRAN (R 3.6.0)
   RCurl                  1.95-4.12 2019-03-04 [1] CRAN (R 3.6.0)
   Rdpack                 0.11-0    2019-04-14 [1] CRAN (R 3.6.0)
   readr                * 1.3.1     2018-12-21 [1] CRAN (R 3.6.0)
   readxl                 1.3.1     2019-03-13 [1] CRAN (R 3.6.0)
   reshape2               1.4.3     2017-12-11 [1] CRAN (R 3.6.0)
   reticulate             1.12      2019-04-12 [1] CRAN (R 3.6.0)
   rlang                  0.3.4     2019-04-07 [1] CRAN (R 3.6.0)
   rmarkdown              1.13      2019-05-22 [1] CRAN (R 3.6.0)
   ROCR                   1.0-7     2015-03-26 [1] CRAN (R 3.6.0)
   rprojroot              1.3-2     2018-01-03 [1] CRAN (R 3.6.0)
   rstudioapi             0.10      2019-03-19 [1] CRAN (R 3.6.0)
   rsvd                   1.0.1     2019-06-02 [1] CRAN (R 3.6.0)
   Rtsne                  0.15      2018-11-10 [1] CRAN (R 3.6.0)
   rvest                  0.3.4     2019-05-15 [1] CRAN (R 3.6.0)
   S4Vectors            * 0.22.0    2019-05-02 [1] Bioconductor  
   scales                 1.0.0     2018-08-09 [1] CRAN (R 3.6.0)
   scater               * 1.12.2    2019-05-24 [1] Bioconductor  
   scran                * 1.12.1    2019-05-27 [1] Bioconductor  
   sctransform            0.2.0     2019-04-12 [1] CRAN (R 3.6.0)
   SDMTools               1.1-221.1 2019-04-18 [1] CRAN (R 3.6.0)
   sessioninfo            1.1.1     2018-11-05 [1] CRAN (R 3.6.0)
   Seurat                 3.0.2     2019-06-14 [1] CRAN (R 3.6.0)
   SingleCellExperiment * 1.6.0     2019-05-02 [1] Bioconductor  
   statmod                1.4.32    2019-05-29 [1] CRAN (R 3.6.0)
   stringi                1.4.3     2019-03-12 [1] CRAN (R 3.6.0)
   stringr              * 1.4.0     2019-02-10 [1] CRAN (R 3.6.0)
   SummarizedExperiment * 1.14.0    2019-05-02 [1] Bioconductor  
 P survival               2.44-1.1  2019-04-01 [5] CRAN (R 3.6.0)
   tibble               * 2.1.3     2019-06-06 [1] CRAN (R 3.6.0)
   tidygraph              1.1.2     2019-02-18 [1] CRAN (R 3.6.0)
   tidyr                * 0.8.3     2019-03-01 [1] CRAN (R 3.6.0)
   tidyselect             0.2.5     2018-10-11 [1] CRAN (R 3.6.0)
   tidyverse            * 1.2.1     2017-11-14 [1] CRAN (R 3.6.0)
   tsne                   0.1-3     2016-07-15 [1] CRAN (R 3.6.0)
   tweenr                 1.0.1     2018-12-14 [1] CRAN (R 3.6.0)
   vipor                  0.4.5     2017-03-22 [1] CRAN (R 3.6.0)
   viridis                0.5.1     2018-03-29 [1] CRAN (R 3.6.0)
   viridisLite            0.3.0     2018-02-01 [1] CRAN (R 3.6.0)
   whisker                0.3-2     2013-04-28 [1] CRAN (R 3.6.0)
   withr                  2.1.2     2018-03-15 [1] CRAN (R 3.6.0)
   workflowr              1.4.0     2019-06-08 [1] CRAN (R 3.6.0)
   xfun                   0.7       2019-05-14 [1] CRAN (R 3.6.0)
   xml2                   1.2.0     2018-01-24 [1] CRAN (R 3.6.0)
   XVector                0.24.0    2019-05-02 [1] Bioconductor  
   yaml                   2.2.0     2018-07-25 [1] CRAN (R 3.6.0)
   zlibbioc               1.30.0    2019-05-02 [1] Bioconductor  
   zoo                    1.8-6     2019-05-28 [1] CRAN (R 3.6.0)

[1] /group/bioi1/luke/analysis/OzSingleCells2019/packrat/lib/x86_64-pc-linux-gnu/3.6.0
[2] /group/bioi1/luke/analysis/OzSingleCells2019/packrat/lib-ext/x86_64-pc-linux-gnu/3.6.0
[3] /group/bioi1/luke/analysis/OzSingleCells2019/packrat/lib-R/x86_64-pc-linux-gnu/3.6.0
[4] /home/luke.zappia/R/x86_64-pc-linux-gnu-library/3.6
[5] /usr/local/installed/R/3.6.0/lib64/R/library

 P ── Loaded and on-disk path mismatch.</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
