---
title: "Cluster comparison"
date: "`r Sys.Date()`"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r knitr, include = FALSE}
source(here::here("R/setup_knitr.R"))
```

```{r setup, cache = FALSE}
#### LIBRARIES ####
# Package conflicts
library("conflicted")

# Single-cell
library("SingleCellExperiment")

# Plotting
library("clustree")

# File paths
library("fs")
library("here")

# Presentation
library("knitr")
library("jsonlite")

# Tidyverse
library("tidyverse")

### CONFLICT PREFERENCES ####
conflict_prefer("path", "fs")
conflict_prefer("rename", "dplyr")

### SOURCE FUNCTIONS ####
source(here("R/output.R"))
source(here("R/crossover.R"))

### OUTPUT DIRECTORY ####
OUT_DIR <- here("output", DOCNAME)
dir_create(OUT_DIR)

#### SET GGPLOT THEME ####
theme_set(theme_minimal())

#### SET PATHS ####
source(here("R/set_paths.R"))
```

Introduction
============

In this document we are going to compare the clustering results for the RNA-seq
and CITE data to see how similar they are to each other.

```{r load, cache.extra = c(tools::md5sum(PATHS$sce_clust), tools::md5sum(PATHS$cite_clust))}
if (all(file_exists(c(PATHS$sce_clust, PATHS$cite_clust)))) {
    sce <- read_rds(PATHS$sce_clust)
    cite <- read_rds(PATHS$cite_clust)
} else {
    stop("Clustered dataset is missing. ",
         "Please run '04-clustering.Rmd' and '05-cite-clustering.Rmd' first.",
         call. = FALSE)
}

clust_data <- colData(sce) %>%
  as.data.frame() %>%
  select(Barcode, GeneCluster = Cluster) %>%
  mutate(AntiCluster = colData(cite)$Cluster)
```

Clustering
==========

Clustering "tree"
-----------------

```{r clustering}
clust_data %>%
  rename(Cluster1 = GeneCluster, Cluster2 = AntiCluster) %>%
  clustree(prefix = "Cluster", show_axis = TRUE) +
  scale_y_continuous(
      breaks = c(0, 1),
      labels = c("CITE cluster", "RNA cluster")
  )
```

Jaccard index
-------------

```{r jaccard}
plot_data <- summariseClusts(clust_data, GeneCluster, AntiCluster) %>%
    replace_na(list(Jaccard = 0))

ggplot(plot_data, aes(x = GeneCluster, y = AntiCluster, fill = Jaccard)) +
    geom_tile() +
    scale_fill_viridis_c(limits = c(0, 1), name = "Jaccard\nindex") +
    coord_equal() +
    labs(
        x = "Gene cluster",
        y = "CITE cluster"
    ) +
    theme(axis.text = element_text(size = 10, colour = "black"),
          axis.ticks = element_blank(),
          axis.title = element_text(size = 15),
          legend.key.height = unit(30, "pt"),
          legend.title = element_text(size = 15),
          legend.text = element_text(size = 10),
          panel.grid = element_blank())
```

Bars {.tabset}
----

### Gene clusters {.unnumbered}

```{r bars-gene}
ggplot(clust_data, aes(x = GeneCluster, fill = AntiCluster)) +
    geom_bar()
```

### CITE clusters {.unnumbered}

```{r bars-cite}
ggplot(clust_data, aes(x = AntiCluster, fill = GeneCluster)) +
    geom_bar()
```

Markers
=======

```{r anti-props}
cite_corr_mat <- logcounts(cite) %>%
    t() %>%
    cor(method = "spearman")

cite_corr_order <- hclust(dist(cite_corr_mat))$order
cite_corr_levels <- rownames(cite_corr_mat)[cite_corr_order]

cite_props <- crossing(
    Cluster = clust_data$AntiCluster,
    Antibody = rownames(cite)
) %>%
    mutate(
        AntiClust = map2_dbl(Cluster, Antibody, function(c, a) {
            sum(counts(cite)[a, clust_data$AntiCluster == c])
        })
    ) %>%
    group_by(Antibody) %>%
    mutate(AntiTotal = sum(AntiClust)) %>%
    group_by(Cluster) %>%
    mutate(ClustTotal = sum(AntiClust)) %>%
    ungroup() %>%
    mutate(
        AntiClustProp = AntiClust / ClustTotal,
        AntiDataProp = AntiTotal / sum(counts(cite)),
        Ratio = AntiClustProp / AntiDataProp
    )

ggplot(cite_props,
       aes(
           x = factor(str_remove(Antibody, "Anti-"),
                      levels = str_remove(cite_corr_levels, "Anti-")),
           y = Ratio, 
           colour = log10(Ratio)
       )) +
    annotate("rect", xmin = -Inf, xmax = Inf, ymin = -Inf, ymax = 2,
             fill = "grey", colour = "grey", alpha = 0.3) +
    geom_hline(yintercept = 2, colour = "red") +
    geom_point() +
    scale_colour_viridis_c() +
    facet_wrap(~ Cluster, ncol = 1, strip.position = "right", scales = "free_y") +
    labs(
        title = "Anitbody proportions",
        y = "(Cluster proportion) / (Dataset proportion)"
    ) +
    theme(
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5),
        legend.position = "none",
        panel.grid.minor = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.border = element_rect(color = "black", fill = NA, size = 1)
    )
```

Summary
=======
Parameters
----------

This table describes parameters used and set in this document.

```{r parameters}
params <- list(
  
)
params <- toJSON(params, pretty = TRUE)
kable(fromJSON(params))
```

Output files
------------

This table describes the output files produced by this document. Right click
and _Save Link As..._ to download the results.

```{r output}
kable(data.frame(
    File = c(
        download_link("parameters.json", OUT_DIR)
    ),
    Description = c(
        "Parameters set and used in this analysis"
    )
))
```

Session information
-------------------

[genecards]: https://www.genecards.org
[novus]: https://www.novusbio.com
